### PR TITLE
3.5.2 Cherry Picks, update puppet modules

### PR DESCRIPTION
**Pull request by akofink**
These cherry-picks had no merge conflicts.

Also included puppet module updates.

### CODE REVIEW

**Comment by akofink** [comment]
@ekohl mind reviewing the puppetfile updates? It's updated to include [this patch in puppet-pulp](https://github.com/theforeman/puppet-pulp/commit/cd36eaee1a56c4c444b57bdf7f8b6ea80ab51230).

**Comment by akofink** [comment]
@mbacovsky mind reviewing your cherry-picked patch?

File: Puppetfile.lock [hunk 1]
```diff
@@ -72,9 +75,9 @@ FORGE
       puppetlabs-apt (< 5.0.0, >= 2.0.0)
       puppetlabs-concat (< 5.0.0, >= 1.1.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.13.1)
-    puppetlabs-puppetdb (6.0.1)
```
Comment (ekohl) [inline Puppetfile.lock]: I'd leave this change out since it's foreman-installer that ships this module.

File: Puppetfile.lock [hunk 2]
```diff
@@ -36,12 +36,12 @@ FORGE
       puppetlabs-apache (< 3.0.0, >= 1.0.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.6.0)
       theforeman-foreman (< 9.0.0, >= 7.0.0)
-    katello-pulp (5.0.1)
+    katello-pulp (5.3.0)
```
Comment (ekohl) [inline Puppetfile.lock]: To keep the changes minimal I'd rather do a 5.0.2 release with that single change. Is it only about that one or are there others?

Comment (akofink) [reply Puppetfile.lock]: Just this change. A 5.0.2 release would be great! :1st_place_medal:

Comment (ekohl) [reply Puppetfile.lock]: https://forge.puppet.com/katello/pulp/5.0.2/readme

**Comment by akofink** [comment]
Updated :)

**Comment by akofink** [comment]
@ekohl I think this is ready for merge. (I don't have permission here)

**Comment by akofink** [comment]
Added a commit to bump the version.

**Comment by akofink** [comment]
@ekohl I think I'm going to need [this change](https://github.com/theforeman/puppet-puppet/commit/cc2971e32163df5e6ebc278bd5e03d5afda72305#diff-bdbcaff63151e0cee35cfece7fdc3cf9R92). Getting the following on upgrade:

```
Exception in thread "main" java.lang.RuntimeException: Service ':FilesystemWatchService' not found
```

**Pull request merged**